x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/restriction/WorkflowResolver.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.restriction;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.Strings;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public final class WorkflowResolver {
+
+    private static final Logger logger = LogManager.getLogger(WorkflowResolver.class);
+
+    /**
+     * Allows access to Search Application query REST API.
+     */
+    public static final Workflow SEARCH_APPLICATION_QUERY_WORKFLOW = Workflow.builder()
+        .name("search_application_query")
+        .addAllowedRestHandlers("search_application_query_action")
```

Comment:
If `RestQuerySearchApplicationAction` changes its name, the value here won't be automatically updated. An alternative is to let each RestHandler declare its (optional) workflow name in addition to the existing `name`. But that would bind workflow tightly to RestHandler and does not seem ideal. Maybe we can just add relevant tests in future PRs to ensure the names here are in sync with the RestHandlers.

Reply:
> Maybe we can just add relevant tests in future PRs to ensure the names here are in sync with the RestHandlers.

As discussed in sync, this is a valid concern and relying on having a relevant tests per workflow should be sufficient to detect name changes.